### ISSUE TITLE
qemu/libvirt issue on CentOS, `virtio-9p-pci` is not valid

### ISSUE THREAD

**Issue by pgburt** [coreos/bugs#1805]
## Bug ##

Question from the [Google Group](https://groups.google.com/forum/#!topic/coreos-user/BXJv4uLCcrk):

> I tried to boot Container Linux on my machine, but I failed whatever use libvirt and qemu by following below instructions.

> The error report showed there don't have 9p installed in my CentOS machines, that seems corrects because all CentOS distribution never build 9p.

> My question is: does 9p is mandatory to boot Container Linux? All I want is just boot the coreos system easily.

### Environment ###

CentOS, local

### Expected Behavior ###

Able to follow qemu or libvirt guides.

### Actual Behavior ###

Get's a warning about `'virtio-9p-pci' is not a valid device model name` 

### Reproduction Steps ###

On CentOS, follow one of the guides. For more info see the question in the [Google Group](https://groups.google.com/forum/#!topic/coreos-user/BXJv4uLCcrk).

**Comment by dm0-** [comment]
The `9p` functionality is only used for the config drive.  If you're using the QEMU wrapper script, adding `-c /dev/null` should prevent it from using `9p`, but then it won't automatically set up any initial configuration.  Just running a QEMU command directly to boot the disk image will probably be the easiest solution.

**Comment by xianwei2016** [comment]
Can you pass me the full command line? or let me know how to change below code section in coreos_production_qemu.sh? great thanks.

```
# ${CONFIG_DRIVE} or ${CONFIG_IMAGE} will be mounted in CoreOS as /media/configdrive
if [ -n "${CONFIG_DRIVE}" ]; then
    set -- \
        -fsdev local,id=conf,security_model=none,readonly,path="${CONFIG_DRIVE}" \
        -device virtio-9p-pci,fsdev=conf,mount_tag=config-2"$@"
fi

if [ -n "${CONFIG_IMAGE}" ]; then
    set -- -drive if=virtio,file="${CONFIG_IMAGE}" "$@"
fi
```

**Comment by dm0-** [comment]
@xianwei2016 Just run `qemu-kvm -cpu host -m 2G -vga std coreos_production_image.bin` or whatever QEMU settings you want to use, or delete the code block you pasted.

**Comment by dalu** [comment]
kubernetes/cluster/libvirt-coreos/user_data.yml
```
        [Mount]
        What=kubernetes
        Where=/opt/kubernetes
        Options=ro,trans=virtio,version=9p2000.L
        Type=9p
```
This line is responsible for that, because qemu/kvm on CentOS 7 doesn't support plan9 resource sharing.

What is the alternative?

**Comment by crawford** [comment]
@dalu You'll need to use a distro that supports 9p. I know that works out of the box on both NixOS and Fedora, but there are probably many others. Alternatively, you could use another method for sharing resources, but that's going to require a fair amount of customization of the configuration.

**Comment by dalu** [comment]
If the prophet doesn't want to come to the mountain - use a different mountain 😆 

Well I can't remotely install a 9p supporting distro. I'll probably have to do nested virtualization for testing purposes.
Ah I see what you did there, instead of downloading the latest release you're just mounting the kubernetes directory into the vm. It's probably also because of access to cloud init for each vm.
An alternative could be to use mkisofs to create the cloud inits and mount them, then download kubernetes release for every guest.
This way the demo would work on all OSes. But guests might fail.

Ok long story short, use a different mountain (OS) or nested virtualization.
Thanks @crawford

**Comment by Asher-Shoshan** [comment]
Is there a way to get it via cloud-init iso image (instead of mounts)

**Comment by crawford** [comment]
@Asher-Shoshan Sure, you could build the ISO and mount the device to the VM but keep in mind that this method has been deprecated. You should instead use Ignition. It uses QEMU's firmware config device to pass the configuration, so 9p support isn't needed.
